--- a/CMakeLists.txt	2019-07-09 14:47:12.552425226 +0200
+++ b/CMakeLists.txt	2019-07-09 14:50:02.595001358 +0200
@@ -143,7 +143,7 @@
 
 # Note: Module CodeCoverage requires the name of the debug build to conform
 #       to cmake standards (first letter uppercase).
-set(BUILD_TYPES Production Debug Testing Competition)
+set(BUILD_TYPES Production Debug Testing Competition Gentoo)
 
 if(ENABLE_ASAN)
 #_cmake_modify_IGNORE   set(CMAKE_BUILD_TYPE Debug)
@@ -166,12 +166,10 @@
 endif()
 
 message(STATUS "Building ${CMAKE_BUILD_TYPE} build")
-include(Config${CMAKE_BUILD_TYPE})
 
 #-----------------------------------------------------------------------------#
 # Compiler flags
 
-add_check_c_cxx_flag("-O${OPTIMIZATION_LEVEL}")
 add_check_c_cxx_flag("-Wall")
 add_check_c_flag("-fexceptions")
 add_check_c_cxx_flag("-Wno-deprecated")
